pkg/controller/bookkeepercluster/bookie.go

```diff
@@ -338,13 +338,19 @@ func MakeBookieConfigMap(bk *v1alpha1.BookkeeperCluster) *corev1.ConfigMap {
 	}
 	gcOpts = util.OverrideDefaultJVMOptions(gcOpts, bk.Spec.JVMOptions.GcOpts)
 
-	gcLoggingOpts := []string{
-		"-XX:+PrintGCDetails",
-		"-XX:+PrintGCDateStamps",
-		"-XX:+PrintGCApplicationStoppedTime",
-		"-XX:+UseGCLogFileRotation",
-		"-XX:NumberOfGCLogFiles=5",
-		"-XX:GCLogFileSize=64m",
+	gcLoggingOpts := []string{}
+	if match, _ := util.CompareVersions(bk.Spec.Version, "0.9.0", "<"); match {
```

Comment:
You are comparing Pravega versions here, not Bookkeeper versions, right? But this is the Bookkeeper Operator and it should be related only to Bookkeeper versions. If you do this kind of checks in other places, they will need to be revisited too.

Is there a more generic and precise way of using the right options? For instance, could you make this condition based on the JDK version of the image?

Reply:
@RaulGracia we are comparing bookkeeper image version here and not the pravega version. Seeing that we have Java 11 from Bookkeeper version 0.9 onwards, this check would work as expected

Reply:
@SrishT I understand this, and this may lead to the conclusion that we should not be creating Bookkeeper images with Pravega versioning, right? Because this pollutes the internal logic of the operator, like in this case. If we agree that this is a Bookkeeper Operator just for Bookkeeper (i.e., independent of Pravega), then you may agree that the version used in this `if` condition has nothing to do with the Bookkeeper versions in the official repo: https://hub.docker.com/r/apache/bookkeeper/tags?page=1&ordering=last_updated

From my point of view, the versions in the link above are the ones that we should be considering, not the ones produced by some arbitrary formatting decision in Pravega. This is making an unnecessary and negative coupling between both projects. You have to keep in mind that a user may want to deploy Bookkeeper alone with this Operator in the same way that happens to the Zookeeper Operator. If we start adding internal logic to the Operator based on Pravega stuff, we are going against this goal.